clinic/schema_based_on_diagram.sql

```diff
@@ -0,0 +1,64 @@
+--Create clinic databased
+CREATE DATABASE clinic
+
+--Create patients table
+CREATE TABLE patients  (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  name VARCHAR(30),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+--Create medical histories table
+CREATE TABLE medical_histories (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  FOREIGN KEY(patient_id) REFERENCES patients(id),
+  status VARCHAR,
+  PRIMARY KEY(id)
+);
+
+--Create treatments table
+CREATE TABLE treatments (
+  id INT GENERATED BY DEFAULT AS IDENTITY,
+  type VARCHAR,
+  name VARCHAR,
+  PRIMARY KEY (id)
+);
+
+--Create invoices table
+CREATE TABLE invoices (
+id INT GENERATED BY DEFAULT AS IDENTITY,
+total_amount DECIMAL,
+generated_at TIMESTAMP,
+payed_at TIMESTAMP,
+medical_history_id INT,
+FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+PRIMARY KEY (id)
+);
+
+--Create invoice items table
+CREATE TABLE invoice_items (
+id INT GENERATED BY DEFAULT AS IDENTITY,
+unit_price DECIMAL,
+quantity INT,
+total_price DECIMAL,
+invoice_id INT,
+treatment_id INT,
+FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+PRIMARY KEY (id)
+);
+
+--Join table for medical history and treatment
+CREATE TABLE treatment_history_link (
+medical_history_id INT,
+treatments_id INT,
+FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+FOREIGN KEY (treatments_id) REFERENCES treatments (id)
+);
+
+--Create indexes for the join table
+CREATE INDEX medical_history_id_asc ON treatment_history_link(medical_history_id ASC)
+CREATE INDEX treatments_id_asc ON treatment_history_link(treatments_id ASC)
```

Comment:
- [ ] It's a requirement that all FOREIGN KEYS have indexes created. Kindly do so:
- Check the image below for the missing indexes
![image](https://user-images.githubusercontent.com/81932478/184976153-4e269f30-9922-429e-9d65-4bfb632a4160.png)

Reply:
Thank you @Munsa1 we implemented the changes.